trin-core/src/portalnet/overlay.rs

```diff
@@ -490,6 +675,24 @@ impl<TContentKey: OverlayContentKey + Send, TMetric: Metric + Send>
     }
 }
 
+/// Randomly select log2 nodes. log2() is stable only for floats, that's why we cast it first to f32
+fn log2_random_enrs(enrs: Vec<Enr>) -> Vec<Enr> {
+    assert!(!enrs.is_empty());
+    // log2(1) is zero but we want to propagate to at least one node
+    if enrs.len() == 1 {
+        return enrs;
+    }
+
+    // Greedy gossip by ceiling the log2 value
+    let log2_value = (enrs.len() as f32).log2().ceil();
+
+    let random_enrs: Vec<Enr> = enrs
+        .clone()
```

Comment:
I'm still slowly building Rust intuition: why do we need to clone here? I would have thought that with ownership of `enrs` we could destroy it with `into_iter()` and not have a problem.

Reply:
Good catch, no need to clone here, I wish clippy can catch those unnecessary clones.